safe-global/web-core
# feat: transaction simulation

## What it solves

Adding transaction simulation

## How this PR fixes it

This introduces the `<TxSimulation>` component used in the `<SignOrExecuteForm>` and `<BatchExecuteModal>` modals and the `useSimulation` hook, based on top of the Core SDK.

Transactions are simulated on Tenderly and the results displayed in the UI.

## How to test it

Create a transaction and observe the simulation option on the review screen of both the standard and batch execution modals.

## Analytics changes

Simulating a transaction dispatches the `MODALS_EVENTS.SIMULATE_TX` event.

## Screenshots

![singular](https://user-images.githubusercontent.com/20442784/187729572-6ecee030-e7e4-412f-9af8-bddb8b44f68c.gif)

![multisend](https://user-images.githubusercontent.com/20442784/187729591-07b13db0-1dbc-4d85-85dd-fb860eca4d5f.gif)
